### ISSUE TITLE
Automate the main bulk of our release tasks

### ISSUE THREAD

**Issue by apyrgio** [freedomofpress/dangerzone#1016]
Our release tasks, as described in our [RELEASE.md](https://github.com/freedomofpress/dangerzone/blob/main/RELEASE.md), take a lot of time to run, mostly for three reasons:
1. Waiting for a command to finish. Some of our commands, such as building images or development environments, take a few minutes to finish. Usually by then, the developer may switch to another task, and check the running command irregularly. The context switching cost and the drift between the command's end adds a lot of time to the release.
2. Commands run sequentially. There are some things that we need to do in order to prepare artifacts for the release (such as building the container image) and other things that can be done in parallel (such as building the Fedora RPMs for each version). Currently, these happen sequentially because the destination dir of the artifacts is the hardcoded `dist/` folder in our repo, and parallel runs may overwrite the contents in it.
3. Environment state. Sometimes our build environment may be in a state where it's not clean, or there's no space. In these cases, we have to debug errors with our artifacts or our commands, which cost us some time.

Automating the main bulk of our release tasks, letting them run in parallel, and having a way to bring our build environment to a clean state will be huge wins, until we have a way to build the artifacts in isolated runners.
